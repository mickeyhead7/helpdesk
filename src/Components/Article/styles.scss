@import "../../themes/default/settings.colors.scss";
@import "../../themes/default/settings.media.scss";

.articleFull {
  .articleHeader {
    @media #{$desktopUp} {
      display: flex;
      align-items: center;

      .articleDate {
        margin: 0 0 0 auto;
      }
    }
  }

  .articleContent {
    background-color: $white;
    margin: 0 0 1em;
    padding: 1em;

    @media #{$desktopUp} {
      margin: 0 0 2em;
      padding: 2em;
    }
  }

  .articleLinks {
    display: flex;
    margin: 0 -1em;

    li {
      list-style: none;
      margin: 0 1em;
      width: 50%;
    }
  }
}
