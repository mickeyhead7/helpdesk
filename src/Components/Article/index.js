import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Link from 'gatsby-link';
import BodyHeader from '@Components/BodyHeader';
import Card from '@packages/patterns/Card';
import I18nProvider from '@packages/Components/i18n/Provider';
import Translate from '@packages/Components/i18n/Translate';
import enGB from './languages/en-GB';

import './styles.scss';

const phrases = {
  'en-GB': enGB,
};

export const Article = ({
  locale,
  frontmatter,
  html,
  prev,
  next,
}) => {
  const { date, title } = frontmatter;

  return (
    <I18nProvider locale={locale} phrases={phrases}>
      <article className="articleFull">
        <BodyHeader>
          <div className="articleHeader">
            <h1>{title}</h1>
            <small className="articleDate">
              <Translate locale={locale} phrase="messages.postedDate" params={{ date }} />
            </small>
          </div>
        </BodyHeader>
        <section
          className="articleContent"
          dangerouslySetInnerHTML={{ __html: html }}
        />
        <section>
          {(prev || next) && (
            <Fragment>
              <h2>
                <Translate locale={locale} phrase="messages.moreArticles" />
              </h2>
              <ul className="articleLinks">
                {prev && (
                  <li>
                    <Card>
                      <h3>
                        <Translate locale={locale} phrase="messages.previousArticle" />
                      </h3>
                      <h4>
                        <Link to={prev.frontmatter.path}>
                          {prev.frontmatter.title}
                        </Link>
                      </h4>
                      <p>
                        {prev.frontmatter.excerpt}
                      </p>
                    </Card>
                  </li>
                )}
                {next && (
                  <li>
                    <Card>
                      <h3>
                        <Translate locale={locale} phrase="messages.nextArticle" />
                      </h3>
                      <h4>
                        <Link to={next.frontmatter.path}>
                          {next.frontmatter.title}
                        </Link>
                      </h4>
                      <p>
                        {next.frontmatter.excerpt}
                      </p>
                    </Card>
                  </li>
                )}
              </ul>
            </Fragment>
          )}
        </section>
      </article>
    </I18nProvider>
  );
};

Article.propTypes = {
  locale: PropTypes.string.isRequired,
  frontmatter: PropTypes.shape({
    date: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
  html: PropTypes.node.isRequired,
  next: PropTypes.shape({
    frontmatter: PropTypes.shape({
      path: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }).isRequired,
  }),
  prev: PropTypes.shape({
    frontmatter: PropTypes.shape({
      path: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }).isRequired,
  }),
};

Article.defaultProps = {
  next: null,
  prev: null,
};

const mapStateToProps = ({ i18n }) => ({
  locale: i18n.locale,
});

export default connect(mapStateToProps)(Article);
