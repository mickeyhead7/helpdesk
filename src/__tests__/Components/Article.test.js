import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Article } from '@Components/Article';

const content = {
  locale: 'en-GB',
  frontmatter: {
    date: '1970-01-01 00:00:00',
    title: 'Article title',
    path: '/article-path',
  },
  html: <div>Some html</div>,
};

test('Article', () => {
  const article = {
    ...content,
    prev: content,
    next: content,
  };
  const component = shallow(<Article {...article} />);

  expect(toJson(component)).toMatchSnapshot();
});
