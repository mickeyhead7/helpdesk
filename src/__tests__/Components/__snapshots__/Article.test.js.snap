// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article 1`] = `
<Connect(Provider)
  locale="en-GB"
  phrases={
    Object {
      "en-GB": Object {
        "messages": Object {
          "moreArticles": "More articles...",
          "nextArticle": "Next article",
          "postedDate": "Posted %{date}",
          "previousArticle": "Previous article",
        },
      },
    }
  }
>
  <article
    className="articleFull"
  >
    <BodyHeader>
      <div
        className="articleHeader"
      >
        <h1>
          Article title
        </h1>
        <small
          className="articleDate"
        >
          <Connect(Translate)
            locale="en-GB"
            params={
              Object {
                "date": "1970-01-01 00:00:00",
              }
            }
            phrase="messages.postedDate"
          />
        </small>
      </div>
    </BodyHeader>
    <section
      className="articleContent"
      dangerouslySetInnerHTML={
        Object {
          "__html": <div>
            Some html
          </div>,
        }
      }
    />
    <section>
      <React.Fragment>
        <h2>
          <Connect(Translate)
            locale="en-GB"
            phrase="messages.moreArticles"
          />
        </h2>
        <ul
          className="articleLinks"
        >
          <li>
            <Card>
              <h3>
                <Connect(Translate)
                  locale="en-GB"
                  phrase="messages.previousArticle"
                />
              </h3>
              <h4>
                <Link
                  to="/article-path"
                >
                  Article title
                </Link>
              </h4>
              <p />
            </Card>
          </li>
          <li>
            <Card>
              <h3>
                <Connect(Translate)
                  locale="en-GB"
                  phrase="messages.nextArticle"
                />
              </h3>
              <h4>
                <Link
                  to="/article-path"
                >
                  Article title
                </Link>
              </h4>
              <p />
            </Card>
          </li>
        </ul>
      </React.Fragment>
    </section>
  </article>
</Connect(Provider)>
`;
